### PR TITLE
feat(vm): add halo2 support

### PR DESCRIPTION
**Pull request by huangzhiran**
- halo2 support
- redefine vm code

### CODE REVIEW

File: output/types/types.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package output
```
Comment (CoderZhi) [inline output/types/types.go:1]: not appropriate to add a subdirectory

File: msg/types/types.go [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package types
```
Comment (CoderZhi) [inline msg/types/types.go:1]: same here

File: vm/vm.go [hunk 3]
```diff
@@ -5,16 +5,28 @@ import (
...
-	"github.com/machinefi/w3bstream-mainnet/vm/instance/manager"
+	"github.com/machinefi/w3bstream-mainnet/vm/server"
+	"github.com/machinefi/w3bstream-mainnet/vm/types"
 	"github.com/pkg/errors"
 )
 
 type Handler struct {
-	instanceMgr *manager.Mgr
+	risc0ServerAddr       string
+	halo2ServerAddr       string
```
Comment (CoderZhi) [inline vm/vm.go:15]: `map[Type]string` ?

File: vm/types/types.go [hunk 4]
```diff
@@ -1,4 +1,4 @@
-package vm
+package types
```
Comment (CoderZhi) [inline vm/types/types.go:1]: same here

File: vm/vm.go (hunk 3 above)
Comment (huangzhiran) [reply vm/vm.go]: changed

File: cmd/node/main.go [hunk 5]
```diff
@@ -29,8 +29,8 @@ func main() {
 	viper.MustBindEnv("CHAIN_ENDPOINT")
 	viper.MustBindEnv("OPERATOR_PRIVATE_KEY")
 
-	vmHandler := vm.NewHandler(viper.Get("RISC0_SERVER_ENDPOINT").(string), viper.Get("HALO2_SERVER_ENDPOINT").(string), viper.Get("PROJECT_CONFIG_FILE").(string))
-	msgHandler := handler.New(vmHandler, viper.Get("CHAIN_ENDPOINT").(string), viper.Get("OPERATOR_PRIVATE_KEY").(string))
+	vmHandler := vm.NewHandler(viper.Get("RISC0_SERVER_ENDPOINT").(string), viper.Get("HALO2_SERVER_ENDPOINT").(string))
```
Comment (CoderZhi) [inline cmd/node/main.go:32]: it is better to use `SetEndpoint(vm.Type, string)`

**Review by CoderZhi — APPROVED** [review]

**Pull request merged**